libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -211,6 +211,32 @@ bool AP_ESC_Telem::get_raw_rpm(uint8_t esc_index, float& rpm) const
     return true;
 }
 
+// pre_arm_check - returns true if all pre-takeoff checks have completed successfully
+bool AP_ESC_Telem::pre_arm_check(char *failure_msg, const uint8_t failure_msg_len, uint32_t esc_check_mask) const
+{
+    if (esc_check_mask == 0) {
+        return true;
+    }
+
+    const uint32_t active_esc_mask = get_active_esc_mask();
+    // Find the ESCs that are both active and need to be checked
+    uint32_t active_and_need_check = esc_check_mask & active_esc_mask;
```

Comment:
I think this is wrong - think you should use the same logic as in takeoff_check.cpp - compare against configured motors. If that doesn't quite work we should unify the two pieces of functionality